Cargo.toml

```diff
@@ -1,7 +1,7 @@
 [package]
 name = "eth_trie_utils"
 description = "Types and utility functions for building/working with partial Ethereum tries."
-version = "0.4.1"
+version = "0.5.0"
```

Comment:
Oh yeah. So I agree, but `Cargo` acts [a bit special](https://doc.rust-lang.org/cargo/reference/semver.html#change-categories) when the `MAJOR` version is `0`, where breaking changes are allowed in `MINOR`. So pretty much all the rules you would apply for the furthest left number shift over to the right (and the same thing happens if `MINOR` is `0`). So in this case, we can treat `MINOR` the same way we would treat `MAJOR`.

> This guide uses the terms "major" and "minor" assuming this relates to a "1.0.0" release or later. Initial development releases starting with "0.y.z" can treat changes in "y" as a major release, and "z" as a minor release. "0.0.z" releases are always major changes. This is because Cargo uses the convention that only changes in the left-most non-zero component are considered incompatible.

I'm not sure when we should bump the major version to `1`. I was thinking maybe once this library is used in live production code that that might be a good time.